File: api-gin/src/routes/produto-routes.go
package routes

import (
	"github.com/gin-gonic/gin"
	controller "tcc-api-gin/src/controllers"
)

func ProdutoRoutes(router *gin.RouterGroup) {
	produto := router.Group("/produtos")
	{
		produto.GET("/:sku", controller.GetProdutoBySKU)
		produto.GET("/categoria/:categoria", controller.GetProdutoByCategoria)
		produto.GET("/", controller.GetAllProdutos)
		produto.GET("/ofertas", controller.GetAllOfertas)
		produto.GET("/destaques", controller.GetAllDestaques)
		produto.GET("/nome", controller.GetProdutoByName)
	}
}

File: api-express/src/routes/produto-routes.js
import { Router } from "express";
import ProdutoController from "../controllers/produto-controller.js";

const produtoRoutes = Router();

produtoRoutes.get('/categoria/:id', ProdutoController.getByCategoria);
produtoRoutes.get('/ofertas', ProdutoController.getAllOfertas);
produtoRoutes.get('/destaques', ProdutoController.getAllDestaques);
produtoRoutes.get('/nome', ProdutoController.getByNome);
produtoRoutes.get('/:sku', ProdutoController.getBySKU);
produtoRoutes.get('/', ProdutoController.getAll);

export default produtoRoutes;